core/src/test/scala/org/apache/spark/FileSuite.scala

```diff
@@ -700,4 +701,38 @@ class FileSuite extends SparkFunSuite with LocalSparkContext {
 
     assert(collectRDDAndDeleteFileBeforeCompute(true).isEmpty)
   }
+
+  test("SPARK-25100: Using KryoSerializer and" +
+      "setting registrationRequired true can lead job failed") {
+    val tempDir = Utils.createTempDir()
+    val inputDir = tempDir.getAbsolutePath + "/input"
```

Comment:
nit: in reality it's just a file instead of dir - may be better to rename the val or create a directory and create a file inside the directory.

Reply:
Thanks, fixed